### PR TITLE
Add new method LoadHTMLFilesRecursively for recursively load template files in a folder

### PR DESCRIPTION
**Pull request by math-nao**
Currently, Gin doesn't handle recursive load of template files in a folder.
This pull request adds a new method called `LoadHTMLFilesRecursively` to load all files matching a list of extension in a specific folder.

### CODE REVIEW

**Comment by appleboy** [comment]
@thinkerou Please help to review this PR. Thanks.

File: gin.go [hunk 1]
```diff
@@ -194,6 +197,26 @@ func (engine *Engine) LoadHTMLFiles(files ...string) {
...
 
+// LoadHTMLFilesRecursively loads recursively a slice of HTML files mathing an extension
+// and associates the result with HTML renderer.
+func (engine *Engine) LoadHTMLFilesRecursively(folder string, extensionsAllowed []string) {
+
+	files := []string{}
+	sort.Strings(extensionsAllowed)
+
+	filepath.Walk(folder, func(path string, info os.FileInfo, err error) error {
+		extension := strings.ToLower(filepath.Ext(path))
```
Comment (thinkerou) [inline gin.go]: Firstly you need to use lower `extensionsAllowed`

File: gin.go [hunk 2]
```diff
@@ -194,6 +197,26 @@ func (engine *Engine) LoadHTMLFiles(files ...string) {
...
+		indexFound := sort.SearchStrings(extensionsAllowed, extension)
+		if indexFound < len(extensionsAllowed) && extensionsAllowed[indexFound] == extension {
+			files = append(files, path)
```
Comment (thinkerou) [inline gin.go]: There are have `sort` and `search` two steps.
why not use the follow code?

```go
for _, v := range extensionsAllowed {
    if strings.ToLower(v) == extension {
        files = append(files, path)
    }
}
```
moreover, `sort.SearchStrings` maybe have problem, see https://github.com/golang/go/issues/25382

**Comment by appleboy** [comment]
@math-nao any updates?

**Comment by thinkerou** [comment]
ref: https://github.com/gin-gonic/gin#html-rendering

**Comment by thinkerou** [comment]
@math-nao please fix conflict and then I merged it, thanks!

**Comment by ganlvtech** [comment]
> ref: https://github.com/gin-gonic/gin#html-rendering
> `router.LoadHTMLGlob("templates/**/*")`

https://stackoverflow.com/questions/26809484/how-to-use-double-star-glob-in-go

golang's `filepath.Glob` doesn't handle `templates/**/*` like other language. It works same as `template/*/*`.

**Comment by unlikezy** [comment]
Why this thread stoped.....

**Comment by thinkerou** [comment]
@math-nao thanks a lot, I will close the pr, and please see #2328

**Pull request closed without merging**